### PR TITLE
Add GitHub Action to deploy to GitHub pages

### LINKED ISSUE #19: Set up continuous integration with GitHub Actions
Include GitHub actions yml file to build and deploy the React app after every merge to master.

Can also include a step to build and run tests the app on every pull request and block merge if build or tests fail.

### PR DESCRIPTION
**Pull request by sunithapatel**
This GitHub action does build, run tests and deploy to GitHub pages only if the branch is master.
It will run the first two steps for pull requests.

Closes #19.

### CODE REVIEW

**Comment by princiya** [comment]
neat 💯 @sunithapatel

**Comment by sunithapatel** [comment]
FYI:  I see that there are some errors to clean up here to get the build to work. I will take a look at these before we merge this PR as this will be used as a check for future PRs. 
I also want to test that it deploys correctly to GitHub Pages before merging this. Once this is done then we can merge this.

**Comment by sunithapatel** [comment]
Okay got it working. It looks like `npm build` treats warnings as errors which is why the build failed initially. Then the one test we had was wrong so I fixed that as well.

I commented out the condition for deploying to master to test out the GitHub pages deploy and that works too, so I put back the condition.

So now it is ready to merge. It just may cause some conflicts with other PRs since I had to change the links in the Header.js component but it should not be too bad.

File: src/App.test.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { render, screen } from '@testing-library/react';
 import App from './App';
 
-test('renders learn react link', () => {
+test('renders Supported by Women Who Code link', () => {
```
Comment (princiya) [inline src/App.test.js:4]: thanks for fixing the test :)

**Pull request merged**